src/mpi/comm/comm_dup.c

```diff
@@ -178,7 +178,6 @@ int MPI_Comm_dup(MPI_Comm comm, MPI_Comm *newcomm)
 	    "**mpi_comm_dup %C %p", comm, newcomm);
     }
 #   endif
-    *newcomm = MPI_COMM_NULL;
```

Comment:
The only problem with this is that we still want to return `MPI_COMM_NULL` in the case where comm creation failed, and the user provided a valid comm handle ptr. Can you modify the patch to instead protect this assignment? Something like:
```
if (newcomm)
    *newcomm = MPI_COMM_NULL;
```